prymer/api/oligo.py

```diff
@@ -103,6 +125,9 @@ class Oligo(OligoLike, Metric["Oligo"]):
     span: Span
     bases: Optional[str] = None
     tail: Optional[str] = None
+    self_any_th: Optional[float] = None
+    self_end_th: Optional[float] = None
+    hairpin_th: Optional[float] = None
```

Comment:
@tfenne per [this comment below](https://github.com/fulcrumgenomics/prymer/pull/60/files#r1790363446), I think these should not be `Optional`. Does that agree with what you have in mind?